src/Illuminate/Queue/Queue.php

```diff
@@ -235,9 +235,14 @@ protected function withCreatePayloadHooks($queue, array $payload)
     {
         if (! empty(static::$createPayloadCallbacks)) {
             foreach (static::$createPayloadCallbacks as $callback) {
-                $payload = array_merge($payload, call_user_func(
-                    $callback, $this->getConnectionName(), $queue, $payload
-                ));
+                $new = $callback($this->getConnectionName(), $queue, $payload);
+                foreach ($new as $key => $value) {
+                    if (is_array($payload[$key] ?? []) && is_array($value)) {
+                        $payload[$key] = array_merge($payload[$key] ?? [], $value);
+                    } else {
+                        $payload[$key] = $value;
```

Comment:
If the most recent callback's payload is not an array, I override the value. This would override the original value even if it were an array.

Is this wanted, or should the new value be wrapped in `[]` and merged with the original value, if it's an array?